src/components/edit-dl-controller.tsx

```diff
@@ -0,0 +1,141 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import React, { FC, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+import {
+	Divider,
+	ModalBody,
+	ModalFooter,
+	ModalHeader,
+	useSnackbar
+} from '@zextras/carbonio-design-system';
+import { difference, xor } from 'lodash';
+import { useTranslation } from 'react-i18next';
+
+import { EditDLComponent } from './edit-dl';
+import { client } from '../network/client';
+
+export type EditDLControllerComponentProps = {
+	email: string;
+	displayName: string;
+	onClose: () => void;
+	onSave: () => void;
+};
+
+export const getMembersPartition = (
+	originalMembers: Array<string>,
+	updatedMembers: Array<string>
+): { membersToAdd: Array<string>; membersToRemove: Array<string> } => ({
+	membersToAdd: difference(updatedMembers, originalMembers),
+	membersToRemove: difference(originalMembers, updatedMembers)
+});
+
+export const EditDLControllerComponent: FC<EditDLControllerComponentProps> = ({
+	email,
+	displayName,
+	onClose,
+	onSave
+}) => {
+	const [members, setMembers] = useState<string[]>([]);
+	const originalMembersRef = useRef<string[]>([]);
+	const [totalMembers, setTotalMembers] = useState<number>(0);
+	const [t] = useTranslation();
+	const createSnackbar = useSnackbar();
+
+	useEffect(() => {
+		client
+			.getDistributionListMembers(email)
+			.then((response) => {
+				setMembers(() => {
+					originalMembersRef.current = response.members;
```

Comment:
There is a reason to set the ref here instead of the same level of setMembers?

Reply:
We need the original list to be always updated before the members state, so that the isDirty memo can run with the updated original list when the state change.
Thinking about this, an alternative could be changing the ref to be a state so that we do not rely on the order of the setter.